http-client/src/main/java/io/airlift/http/client/StreamingBodyGenerator.java

```diff
@@ -0,0 +1,36 @@
+package io.airlift.http.client;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import static java.util.Objects.requireNonNull;
+
+public class StreamingBodyGenerator
```

Comment:
Update the Javadoc in `BodyGenerator` to mention this class.

Reply:
We should probably seal the interface, as a separate change.

Reply:
https://github.com/airlift/airlift/issues/1181